### PR TITLE
Apply `black` to remaining parts of `astropy.io`

### PR DESCRIPTION
**Pull request by WilliamJamieson**
### Description

Apply `black` to remaining parts of `astropy.io` according to the process laid out by [APE 20](https://github.com/astropy/astropy-APEs/blob/main/APE20.rst).

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [ ] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [ ] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [ ] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [ ] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [ ] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [ ] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Review by nstarman — APPROVED** [review]
Thanks @WilliamJamieson

File: astropy/io/registry/base.py [hunk 1]
```diff
@@ -87,12 +88,16 @@ def get_formats(self, data_class=None, filter_on=None):
...
             *[self._registries[k]["column"] for k in self._registries_order],
-            "Deprecated")
+            "Deprecated",
+        )
         i_dataclass = colnames.index("Data class")
         i_format = colnames.index("Format")
-        i_regstart = colnames.index(self._registries[self._registries_order[0]]["column"])
+        i_regstart = colnames.index(
+            self._registries[self._registries_order[0]]["column"]
+        )
```
Comment (nstarman) [inline astropy/io/registry/base.py:100]: Lines like this are an encouragement to write cleaner code.

File: astropy/io/registry/base.py [hunk 2]
```diff
@@ -238,9 +255,10 @@ def my_identifier(*args, **kwargs):
         if not (data_format, data_class) in self._identifiers or force:
             self._identifiers[(data_format, data_class)] = identifier
         else:
-            raise IORegistryError("Identifier for format '{}' and class '{}' is "
-                                  'already defined'.format(data_format,
-                                                           data_class.__name__))
+            raise IORegistryError(
+                "Identifier for format '{}' and class '{}' is "
+                'already defined'.format(data_format, data_class.__name__)
```
Comment (WilliamJamieson) [inline astropy/io/registry/base.py:260]: Should we convert this (and the rest of the ones modified) `.format` statement into an f-string?

Comment (nstarman) [reply astropy/io/registry/base.py:260]: I think so. I wonder why pyupgrade isn't doing this.

**Comment by nstarman** [comment]
pre-commit.ci autofix

File: astropy/io/registry/core.py [hunk 3]
```diff
@@ -98,9 +100,11 @@ def register_reader(self, data_format, data_class, function, force=False,
         if not (data_format, data_class) in self._readers or force:
             self._readers[(data_format, data_class)] = function, priority
         else:
-            raise IORegistryError("Reader for format '{}' and class '{}' is "
-                              'already defined'
-                              ''.format(data_format, data_class.__name__))
+            raise IORegistryError(
+                "Reader for format '{}' and class '{}' is already defined".format(
+                    data_format, data_class.__name__
```
Comment (nstarman) [inline astropy/io/registry/core.py:105]: Flagging for f-stringification

File: astropy/io/registry/core.py [hunk 4]
```diff
@@ -120,8 +124,11 @@ def unregister_reader(self, data_format, data_class):
         if (data_format, data_class) in self._readers:
             self._readers.pop((data_format, data_class))
         else:
-            raise IORegistryError("No reader defined for format '{}' and class '{}'"
-                                  ''.format(data_format, data_class.__name__))
+            raise IORegistryError(
+                "No reader defined for format '{}' and class '{}'".format(
+                    data_format, data_class.__name__
```
Comment (nstarman) [inline astropy/io/registry/core.py:129]: Same

File: astropy/io/registry/core.py [hunk 5]
```diff
@@ -151,7 +158,9 @@ def get_reader(self, data_format, data_class):
             raise IORegistryError(
                 "No reader defined for format '{}' and class '{}'.\n\nThe "
                 "available formats are:\n\n{}".format(
-                    data_format, data_class.__name__, format_table_str))
+                    data_format, data_class.__name__, format_table_str
```
Comment (nstarman) [inline astropy/io/registry/core.py:161]: Same.

File: astropy/io/registry/core.py [hunk 6]
```diff
@@ -216,8 +228,11 @@ def read(self, cls, *args, format=None, cache=False, **kwargs):
                 try:
                     data = cls(data)
                 except Exception:
-                    raise TypeError('could not convert reader output to {} '
-                                    'class.'.format(cls.__name__))
+                    raise TypeError(
+                        'could not convert reader output to {} class.'.format(
+                            cls.__name__
```
Comment (nstarman) [inline astropy/io/registry/core.py:233]: Same

File: astropy/io/registry/core.py [hunk 7]
```diff
@@ -237,12 +253,17 @@ def __init__(self):
         super().__init__()
         self._writers = OrderedDict()
         self._registries["write"] = dict(attr="_writers", column="Write")
-        self._registries_order = ("write", "identify", )
+        self._registries_order = (
+            "write",
+            "identify",
```
Comment (nstarman) [inline astropy/io/registry/core.py:258]:
```suggestion
            "identify"
```

File: astropy/io/registry/core.py [hunk 8]
```diff
@@ -267,9 +288,11 @@ def register_writer(self, data_format, data_class, function, force=False, priori
         if not (data_format, data_class) in self._writers or force:
             self._writers[(data_format, data_class)] = function, priority
         else:
-            raise IORegistryError("Writer for format '{}' and class '{}' is "
-                                  'already defined'
-                                  ''.format(data_format, data_class.__name__))
+            raise IORegistryError(
+                "Writer for format '{}' and class '{}' is already defined".format(
+                    data_format, data_class.__name__
```
Comment (nstarman) [inline astropy/io/registry/core.py:293]: F-string

File: astropy/io/registry/core.py [hunk 9]
```diff
@@ -289,8 +312,11 @@ def unregister_writer(self, data_format, data_class):
         if (data_format, data_class) in self._writers:
             self._writers.pop((data_format, data_class))
         else:
-            raise IORegistryError("No writer defined for format '{}' and class '{}'"
-                                  ''.format(data_format, data_class.__name__))
+            raise IORegistryError(
+                "No writer defined for format '{}' and class '{}'".format(
+                    data_format, data_class.__name__
```
Comment (nstarman) [inline astropy/io/registry/core.py:317]: Same

File: astropy/io/registry/core.py [hunk 10]
```diff
@@ -320,7 +346,9 @@ def get_writer(self, data_format, data_class):
             raise IORegistryError(
                 "No writer defined for format '{}' and class '{}'.\n\nThe "
                 "available formats are:\n\n{}".format(
-                    data_format, data_class.__name__, format_table_str))
+                    data_format, data_class.__name__, format_table_str
```
Comment (nstarman) [inline astropy/io/registry/core.py:349]: Same

File: astropy/io/registry/core.py (hunk 7 above)
Comment (nstarman) [reply astropy/io/registry/core.py:258]: Hopefully that will collapse the line

**Comment by WilliamJamieson** [comment]
@nstarman if you think this PR is ready for string normalization, please move it to the "Ready for string normalization" column in https://github.com/orgs/astropy/projects/10.

File: astropy/io/registry/interface.py [hunk 11]
```diff
@@ -78,8 +83,9 @@ def help(self, format=None, out=None):
         else:
             if format:
                 # Format-specific
-                header = ("{}.{}(format='{}') documentation\n"
-                          .format(cls.__name__, method_name, format))
+                header = "{}.{}(format='{}') documentation\n".format(
+                    cls.__name__, method_name, format
+                )
```
Comment (nstarman) [inline astropy/io/registry/interface.py:88]:
```suggestion
                header = "{cls.__name__}.{method_name}(format='{format}') documentation\n"
```

File: astropy/io/registry/base.py [hunk 12]
```diff
@@ -238,9 +255,11 @@ def my_identifier(*args, **kwargs):
         if not (data_format, data_class) in self._identifiers or force:
             self._identifiers[(data_format, data_class)] = identifier
         else:
-            raise IORegistryError("Identifier for format '{}' and class '{}' is "
-                                  'already defined'.format(data_format,
-                                                           data_class.__name__))
+            raise IORegistryError(
+                "Identifier for format '{}' and class '{}' is already defined".format(
+                    data_format, data_class.__name__
+                )
```
Comment (nstarman) [inline astropy/io/registry/base.py:261]:
```suggestion
                "Identifier for format {data_format!r} and class {data_class.__name__!r} is already defined"
```

File: astropy/io/registry/base.py [hunk 13]
```diff
@@ -256,8 +275,11 @@ def unregister_identifier(self, data_format, data_class):
         if (data_format, data_class) in self._identifiers:
             self._identifiers.pop((data_format, data_class))
         else:
-            raise IORegistryError("No identifier defined for format '{}' and class"
-                                  " '{}'".format(data_format, data_class.__name__))
+            raise IORegistryError(
+                "No identifier defined for format '{}' and class '{}'".format(
+                    data_format, data_class.__name__
+                )
```
Comment (nstarman) [inline astropy/io/registry/base.py:281]:
```suggestion
                "No identifier defined for format {data_format!r} and class {data_class.__name__!r}"
```

File: astropy/io/registry/base.py [hunk 14]
```diff
@@ -334,11 +357,13 @@ def _get_valid_format(self, mode, cls, path, fileobj, args, kwargs):
...
-                                  " file name or contents, please provide a"
-                                  " 'format' argument.\n"
-                                  "The available formats are:\n"
-                                  "{}".format(format_table_str))
+            raise IORegistryError(
+                "Format could not be identified based on the"
+                " file name or contents, please provide a"
+                " 'format' argument.\n"
+                "The available formats are:\n"
+                "{}".format(format_table_str)
```
Comment (nstarman) [inline astropy/io/registry/base.py:365]:
```suggestion
                f"The available formats are:\n{format_table_str}"
```

**Review by nstarman — CHANGES_REQUESTED** [review]

**Comment by nstarman** [comment]
Other than some straggling f-strings, LGTM! Moving to "Ready for string normalization"

**Comment by WilliamJamieson** [comment]
> Other than some straggling f-strings, LGTM! Moving to "Ready for string normalization"

Made your changes and then normalized the strings.

I also squashed... I got ahead of myself there. Please move the "Ready for merge" if there are no other issues.

**Review by nstarman — APPROVED** [review]

**Review by nstarman — APPROVED** [review]

**Pull request merged**